File: packages/js-sdk/src/regex/detectors/no/orgnr.ts
// Norwegian Organisasjonsnummer (organization number) detector

import { RegexDetector } from '../../base'
import { EntityType } from '../../entities'
import { registerRegion } from '../../registry'
import { noOrgnrChecksum } from '../../validators'

class NoOrgnrDetector extends RegexDetector {
  entityType = EntityType.NO_ORGNR
  score = 0.85
  contextKeywords = ['organisasjonsnummer', 'orgnr', 'org nr', 'organization number', 'foretaksnummer']
  contextRequired = true

  pattern = /\b\d{9}\b/g
  validator = noOrgnrChecksum
}

registerRegion('no', NoOrgnrDetector)

export { NoOrgnrDetector }

File: packages/python-sdk/src/blindfold/regex/detectors/no/orgnr.py
"""Norwegian Organisasjonsnummer (organization number) detector"""

import re

from ...base import RegexDetector
from ...entities import EntityType
from ...registry import register_region
from ...validators import no_orgnr_checksum


@register_region("no")
class OrgnrDetector(RegexDetector):
    entity_type = EntityType.NO_ORGNR
    score = 0.85
    context_keywords = [
        "organisasjonsnummer", "orgnr", "org nr", "organization number",
        "foretaksnummer",
    ]
    context_required = True
    context_window = 50

    # 9-digit organization number
    pattern = re.compile(
        r"\b\d{9}\b"
    )
    validator = staticmethod(no_orgnr_checksum)
